src/kernel/kernel.h

```diff
@@ -567,6 +567,14 @@ static inline void clear_fault_handler(void)
         }                                                               \
     } while (0);
 
+
+#define contextual_closure_init(__name, __var, ...)                                 \
```

Comment:
This could be achieved via an init_closure() and a call to apply_context_to_closure(), as used in iov_op(). Though I think this looks nicer...maybe change iov_op() to use this so there aren't two ways to init a closure with context?

Reply:
OK, I replaced the existing call to apply_context_to_closure() with the use of the new contextual_closure_init() macro.